### ISSUE TITLE
REST endpoints for users

### ISSUE THREAD

**Issue by jakobpalmborg** [fungover/mmotodo#11]
Users:
Get information about a user
Create a new user
Update/Change user information
Delete an existing user
Get information about the user:

GET /api/users/{id}
Create a new user:

POST /api/users - that accepts a JSON object
Change user information:

PUT/PATCH /api/users/{id} - that accepts a JSON object
Delete an existing user:

DELETE /api/users/{id}

**Comment by holst9559** [comment]
Should we add an optional query param to GET?
Similar to the endpoint for tasks. We filter out users who belong to a specific team. And do we want to filter based on multiple teams?In case we want to find what users who have more than 1 team in common?

GET /api/users?teamId=id
Should return all user belonging to one team

GET /api/users?teamIdOne={idOne}&teamIdTwo={idTwo}
Should return all user belonging to both teams

Maybe it's an unnecessary endpoint since we got the Teams endpoint to get the members from teams.

**Comment by saeraa** [comment]
@holst9559 
Wouldn't that be better left to the GraphQL queries we'll be using later?

**Comment by holst9559** [comment]
I agree, it would make for cleaner code in the endpoints. Let's leave it to the GraphQL queries.
